### ISSUE TITLE
Manually Building up Pipeline Doesn't Work?

### ISSUE THREAD

**Issue by r-clancy** [explosion/spaCy#3048]
## How to reproduce the behaviour
```
from spacy.lang.en import English

# This version works
segmenter = spacy.load("en")
segmenter.disable_pipes("parser", "tagger")
segmenter.add_pipe(segmenter.create_pipe("sentencizer"), first=True)

# This version doesn't
segmenter = English()
segmenter.add_pipe(segmenter.create_pipe("sentencizer"), first=True)
segmenter.add_pipe(segmenter.create_pipe("ner"))
```

The following error is thrown when attempting to use the second version:
```
Traceback (most recent call last):
  File "/home/ryan1clancy/bigdata2018f-project/src/main.py", line 49, in <module>
    ner(paragraphs)
  File "/home/ryan1clancy/bigdata2018f-project/src/spacy_ner.py", line 35, in ner
    parsed = segmenter("Apple is a company. They are from California.")
  File "/home/ryan1clancy/.conda/envs/bigdata2018f/lib/python3.7/site-packages/spacy/language.py", line 346, in __call__
    doc = proc(doc)
  File "nn_parser.pyx", line 338, in spacy.syntax.nn_parser.Parser.__call__
  File "nn_parser.pyx", line 401, in spacy.syntax.nn_parser.Parser.parse_batch
  File "nn_parser.pyx", line 728, in spacy.syntax.nn_parser.Parser.get_batch_model
TypeError: 'bool' object is not iterable
```

Interestingly, if I only have one of the `add_pipe` calls in the second version it seems works - it only fails when both are used.

As an aside, is this the proper way to achieve what I want? I want to perform NER on each sentence in the paragraph and have the start/end indexes based on the sentence rather than paragraph.

Let's say I'm working with this code:

```
    parsed = segmenter("Apple is a company. They are from California.")
    for sent in parsed.sents:
        for ent in sent.ents:
            print(ent.text, ent.label_, ent.start_char, ent.end_char)
```

It currently prints:
```
Apple ORG 0 5
California GPE 34 44
```

but I want something like
```
Apple ORG 0 5
California GPE 14 23
```

where the `start_char` and `end_char` are based on the sentence they're in and not the paragraph.

## Info about spaCy

* **spaCy version:** 2.0.18
* **Platform:** Linux-4.15.0-39-generic-x86_64-with-debian-buster-sid
* **Python version:** 3.7.1
* **Models:** en

**Comment by ines** [comment]
```python
# This version doesn't
segmenter = English()
segmenter.add_pipe(segmenter.create_pipe("sentencizer"), first=True)
segmenter.add_pipe(segmenter.create_pipe("ner"))
```

I think the problem here is that the `English()` language class doesn't have any weights initialized. This causes the cryptic "'bool' object is not iterable" error (which will be fixed in the upcoming 2.1.0 btw!). 

Try calling `nlp.begin_training()` after adding the pipes and see if it solves the problem!

> As an aside, is this the proper way to achieve what I want? I want to perform NER on each sentence in the paragraph and have the start/end indexes based on the sentence rather than paragraph.

Not really, no. spaCy always operates on the whole document. So your pipeline would currently only split the text into sentences (and make them available via the `doc.sents`) and then also predict named entities. Except in the second version, where you're using only the `English` language class class with no statistical model loaded and no pre-trained weights for the named entity recognizer. In that code, the entity recognizer wouldn't predict anything.

If I understand your use case correctly, it might actually be easiest to use the default pipeline and then compute the entity start and end characters relative to the sentence. Named entities in `doc.ents` are `Span` objects, and the `Span.sent` attribute ([see here](https://spacy.io/api/span#attributes)) returns the sentence a span is part of. That sentence will also have a `start_char` and `end_char`, relative to the parent document. So you could do something like this:

```python
for ent in doc.ents:
   relative_start_char = ent.start_char - ent.sent.start_char
   relative_end_char = ent.end_char - ent.sent.start_char
   print(ent.text, ent.label_, relative_start_char, relative_end_char)
```

If you want this to be more elegant, you could also use [custom extension attributes](https://spacy.io/usage/processing-pipelines#custom-components-attributes) to make these available as something like `ent._.relative_start_char` and `ent._.relative_end_char`.

**Comment by honnibal** [comment]
Better errors now raised by #3074
